### PR TITLE
README.md - Add Table Of Contents+Guide for Docker

### PR DESCRIPTION
**Pull request by benjisho**
### Background

1. There was no Table of Contents so I made one for reading the documentation more optimally
2. There was no link that redirects on how to install Docker Engine - for people who just starting with Docker.

### Changes

1. Added a Table of Contents for easy navigation through the README.md.
2. Included a new "Installing Docker" section with a link to the Official Docker Engine Installation Guide.

### Documentation

For example the "Features" topic is written in the following way:
In the Table of contents:
`[🚀 Features](#features)`

And where the actual topic is located i added a link with the variable:
`## <a name="features"></a>🚀 Features`

### Test Plan

**- no tests needed, just document edits.**

### PR Quality Checklist
- [v] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts. **- no tests needed, just document edits.**
- [ ] I have considered potential risks and mitigations for my changes. **- no risk, just document edits.**
- [v] I have documented my changes clearly and comprehensively.
- [v] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,32 +51,72 @@ Development of this free, open-source project is made possible by all the <a hre
...
+- [GPT3.5 ONLY Mode](#gpt35-only-mode)
+- [🖼 Image Generation](#image-generation)
+- [Selenium](#selenium)
+- [⚠️ Limitations](#limitations)
+- [🛡 Disclaimer](#disclaimer)
+- [🐦 Connect with Us on Twitter](#connect-with-us-on-twitter)
+- [Run tests](#run-tests)
+- [Run linter](#run-linter)
+
+----
```
Comment (Pwuts) [inline README.md]: TOC was removed on purpose as GitHub provides a built-in TOC that doesn't need to be updated manually.

File: README.md [hunk 2]
```diff
@@ -171,15 +211,17 @@ To print out debug logs:
...
 You can also build this into a docker image and run it:
 
 ```bash
 docker build -t autogpt .
 docker run -it --env-file=./.env -v $PWD/auto_gpt_workspace:/app/auto_gpt_workspace autogpt
 ```
-
+#### Installing Docker
+- To install Docker on your machine please follow the [Official Docker Engine Installation Guide](https://docs.docker.com/engine/install/)
+
```
Comment (Pwuts) [inline README.md]: Please move to below `docker-compose` instructions.

Comment (benjisho) [reply README.md]: Moved accordingly

File: README.md (hunk 1 above)
Comment (benjisho) [reply README.md]: Interesting! But I couldn't find a built-in TOC option for README.md files, can you please point me?

Comment (Pwuts) [reply README.md]: image

File: README.md [hunk 3]
```diff
@@ -171,15 +211,15 @@ To print out debug logs:
...
+### <a name="docker"></a>Docker
 
 You can also build this into a docker image and run it:
 
 ```bash
 docker build -t autogpt .
 docker run -it --env-file=./.env -v $PWD/auto_gpt_workspace:/app/auto_gpt_workspace autogpt
 ```
-
+
```
Comment (Pwuts) [inline README.md:182]: trailing whitespace ✂️

File: README.md (hunk 1 above)
Comment (benjisho) [reply README.md]: Cool good to know!

Comment (Pwuts) [reply README.md]: Please remove it from your PR as we removed it intentionally

**Comment by richbeales** [comment]
We have now restructed the README into /docs (it was getting too long).  Please resolve the conflicts/add your docker instructions into the right (new) place.

**Comment by benjisho** [comment]
> We have now restructed the README into /docs (it was getting too long). Please resolve the conflicts/add your docker instructions into the right (new) place.

Done

**Comment by Pwuts** [comment]
This has already been fixed. Thanks for submitting though!

**Pull request closed without merging**